# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/test/test_text-fragments-create-text-directive.html
# Commit: 09616085d8ef
# Full Hash: 09616085d8ef6470309e1b3c565242d6a811cd5d
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-22 09:32:06
# Regressor Bug: 1978235
# File Overlap Count: 1
# Description:
#   Bug 1978235 - Text Fragments: Ensure that start and end terms are always non-empty. r=mccr8,dom-core
#   
#   The patch for Bug 1976473 changed  the behavior of the algorithm to collect word boundary distances
#   to merge punctuation characters with the preceding word. However, this led to unexpected behavior if a
#   context term only consisted of punctuation, where this function would unexpectedly return an empty array.
# ==============================================================================

diff -r df6c5c98b663 -r 09616085d8ef dom/base/test/test_text-fragments-create-text-directive.html
--- a/dom/base/test/test_text-fragments-create-text-directive.html	Mon Jul 21 18:33:07 2025 +0000
+++ b/dom/base/test/test_text-fragments-create-text-directive.html	Mon Jul 21 18:36:11 2025 +0000
@@ -50,6 +50,8 @@
     <span id="afterEmptyTextNode">after empty text node</span>
   </p>
   <div id="nbsp">&nbsp;</div>
+  <div id="punctuationStartsRangeBasedTextDirective">"</div>
+  <div id="punctuationEndsRangeBasedTextDirective">text preceded by punctuation in a different block</div>
   <script>
     document.addEventListener("DOMContentLoaded", () => {
       const empty = document.createTextNode("");
@@ -354,6 +356,15 @@
           content: "node",
           textDirective: "text=node"
         },
+        {
+          name: "Range-based text directive with punctuation as start",
+          startContainer: punctuationStartsRangeBasedTextDirective.firstChild,
+          startOffset: 0,
+          endContainer: punctuationEndsRangeBasedTextDirective.firstChild,
+          endOffset: 4,
+          content: '"\n  text',
+          textDirective: "text=%22,text"
+        }
       ]) {
         const range = document.createRange();
         range.setStart(testCase.startContainer, testCase.startOffset);
